src/main/kotlin/de/bund/digitalservice/useid/config/WidgetSecurityHeadersFilter.kt

```diff
@@ -2,36 +2,37 @@ package de.bund.digitalservice.useid.config
 
 import de.bund.digitalservice.useid.tenant.REQUEST_ATTR_TENANT
 import de.bund.digitalservice.useid.tenant.Tenant
+import de.bund.digitalservice.useid.widget.FALLBACK_PAGE
 import de.bund.digitalservice.useid.widget.WIDGET_PAGE
 import jakarta.servlet.FilterChain
 import jakarta.servlet.http.HttpServletRequest
 import jakarta.servlet.http.HttpServletResponse
 import org.springframework.http.HttpHeaders
 import org.springframework.http.server.PathContainer
 import org.springframework.web.filter.OncePerRequestFilter
-import org.springframework.web.util.pattern.PathPattern
 import org.springframework.web.util.pattern.PathPatternParser
 import java.util.UUID
 
 internal const val HTTP_HEADER_CONTENT_SECURITY_POLICY = "Content-Security-Policy"
 class WidgetSecurityHeadersFilter : OncePerRequestFilter() {
 
-    private val widgetPagePath: PathPattern = PathPatternParser().parse("/$WIDGET_PAGE")
-    private val listOfPages = listOf(widgetPagePath)
-
     override fun doFilterInternal(request: HttpServletRequest, response: HttpServletResponse, filterChain: FilterChain) {
-        val pathIsValidWidgetPages = listOfPages.any {
-            it.matches(PathContainer.parsePath(request.servletPath))
+        if (requestPathMatches(WIDGET_PAGE, request)) {
+            val tenant = request.getAttribute(REQUEST_ATTR_TENANT) as Tenant
+            tenant.cspNonce = UUID.randomUUID().toString()
+            response.setHeader(HTTP_HEADER_CONTENT_SECURITY_POLICY, WidgetContentSecurityPolicy.headerValue(tenant.cspHost, tenant.cspNonce))
+            response.setHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, tenant.cspHost)
+            response.setHeader(HttpHeaders.VARY, HttpHeaders.ORIGIN)
         }
 
-        if (!pathIsValidWidgetPages) return filterChain.doFilter(request, response)
-
-        val tenant = request.getAttribute(REQUEST_ATTR_TENANT) as Tenant
-        tenant.cspNonce = UUID.randomUUID().toString()
-        response.setHeader(HTTP_HEADER_CONTENT_SECURITY_POLICY, WidgetContentSecurityPolicy.headerValue(tenant.cspHost, tenant.cspNonce))
-        response.setHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, tenant.cspHost)
-        response.setHeader(HttpHeaders.VARY, HttpHeaders.ORIGIN)
+        if (requestPathMatches(FALLBACK_PAGE, request)) {
+            response.setHeader(HTTP_HEADER_CONTENT_SECURITY_POLICY, WidgetContentSecurityPolicy.headerValueNoFrame())
```

Comment:
is there a reason why there are no tests for that in WidgetControllerIntegrationTest?

Reply:
Haha, I was looking everywhere for tests for this file but couldn't find any. 😆 Thanks for pointing me to that, of course I want to add a test! :)